bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/ModuleImpl.java

```diff
@@ -0,0 +1,159 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.automation.core.internal;
+
+import org.eclipse.smarthome.automation.Module;
+import org.eclipse.smarthome.automation.type.Input;
+import org.eclipse.smarthome.automation.type.ModuleType;
+import org.eclipse.smarthome.automation.type.Output;
+import org.eclipse.smarthome.config.core.ConfigDescriptionParameter;
+import org.eclipse.smarthome.config.core.Configuration;
+
+/**
+ * Modules are building components of the {@link RuleImpl}s. Each ModuleImpl is
+ * identified by id, which is unique in scope of the {@link RuleImpl}. It also has a {@link ModuleType} which provides
+ * meta
+ * data of the module. The meta data
+ * defines {@link Input}s, {@link Output}s and {@link ConfigDescriptionParameter}s parameters of the {@link ModuleImpl}.
+ * <br>
+ * Setters of the module don't have immediate effect on the RuleImpl. To apply the
+ * changes, they should be set on the {@link RuleImpl} and the RuleImpl has to be
+ * updated by RuleManager
+ *
+ * @author Yordan Mihaylov - Initial Contribution
+ *
+ */
+public abstract class ModuleImpl implements Module {
+
+    /**
+     * Id of the ModuleImpl. It is mandatory and unique identifier in scope of the {@link RuleImpl}. The id of the
+     * {@link ModuleImpl} is used to identify the module
+     * in the {@link RuleImpl}.
+     */
+    protected String id;
```

Comment:
There are getters and setters, so why do you use "protected" members instead of private ones?

Reply:
I agree, will update the pull request.